src/coreclr/utilcode/util.cpp

```diff
@@ -1070,7 +1073,10 @@ int GetCurrentProcessCpuCount()
             {
                 DWORD cpuLimit = (maxRate * GetTotalProcessorCount() + MAXIMUM_CPU_RATE - 1) / MAXIMUM_CPU_RATE;
                 if (cpuLimit < count)
+                {
                     count = cpuLimit;
+                    g_currentProcessIsCpuQuotaLimited = true;
```

Comment:
The `cpuLimit` above is rounded up, so this wouldn't identify some cases where the CPU quota is limited. For example, if the system has one processor and the process is configured to use half of one processor, `cpuLimit` and `count` would be 1, but the CPU quota is actually limited.

I think `cpuLimit` could be calculated as a `double` without any rounding and compared with `count` for determining whether the CPU quota is limited. For determining the CPU count it can round it up and do an int comparison with `count` similarly to here. Similarly for the Unix part, `PAL_GetCpuLimit` could be modified to provide a `double cpuLimit` and the rounding could be done separately. Similarly for NativeAOT and elsewhere as well.